folding-schemes/Cargo.toml

```diff
@@ -23,9 +23,13 @@ num-integer = "0.1"
 color-eyre = "=0.6.2"
 ark-bn254 = {version="0.4.0"}
 ark-groth16 = { version = "^0.4.0" }
+ark-noname = { git = "https://github.com/dmpierre/ark-noname", branch="feat/sonobe-integration" }
```

Comment:
This branch makes it possible to change a noname r1cs into an arkworks one, while taking into account the fact that the `cs` has already been assigned witnesses and public io.

Reply:
How do we deal with this in the long run? PR to noname? An adaptor lib? Feels a bit weak to have this in a random repo

Reply:
We already merged [a PR](https://github.com/zksecurity/noname/pull/137) in noname, which integrates this adaptor library into it directly. But there are two issues:
- Noname is using `arkworks=3.0` and an upgrade to `4.0` might not happen, [for now](https://github.com/zksecurity/noname/issues/138).
- We need to have an adapted version of this lib, accommodating the fact that the witness and public i/o is assigned *before* calling the lib.
This approach of having a fork of an adaptor lib is [similar to what we are doing with `ark-circom`](https://github.com/privacy-scaling-explorations/sonobe/blob/b5667968f48feacbccb1c3f5407f00f7e4892a4a/folding-schemes/Cargo.toml#L18). However, in this case, I hope to only fork `noname` in the future and get rid of `ark-noname`, as soon as `noname` moves on to `arkworks=4.0`.